src/software/ai/hl/stp/tactic/dribble/dribble_fsm.h

```diff
@@ -0,0 +1,304 @@
+         */
+        const auto get_possession = [this](auto event) {
+            auto ball_position = event.common.world.ball().position();
+            auto face_ball_orientation =
+                (ball_position - event.common.robot.position()).orientation();
+            Point intercept_position =
+                findInterceptionPoint(event.common.robot, event.common.world.ball(),
+                                      event.common.world.field());
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), intercept_position, face_ball_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        /**
+         * Action to dribble the ball
+         *
+         * This action will orient the robot towards the destination, dribble to the
+         * destination, and then pivot to face the expected orientation
+         *
+         * @param event DribbleFSM::Update
+         */
+        const auto dribble = [this, &continuous_dribbling_start_point](auto event) {
+            Point ball_position = event.common.world.ball().position();
+            auto [target_destination, target_orientation] =
+                calculateNextDribbleDestinationAndOrientation(
+                    event.common.world.ball(), event.common.robot,
+                    event.control_params.dribble_destination,
+                    event.control_params.final_dribble_orientation);
+            AutoChipOrKick auto_chip_or_kick = AutoChipOrKick{AutoChipOrKickMode::OFF, 0};
+
+            if (!event.control_params.allow_excessive_dribbling &&
+                !comparePoints(ball_position, continuous_dribbling_start_point,
+                               MAX_CONTINUOUS_DRIBBLING_DISTANCE))
+            {
+                // give the ball a little kick
+                auto_chip_or_kick =
+                    AutoChipOrKick{AutoChipOrKickMode::AUTOKICK, DRIBBLE_KICK_SPEED};
+            }
+
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), target_destination, target_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW, auto_chip_or_kick,
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        const auto start_dribble = [this, &continuous_dribbling_start_point,
+                                    dribble](auto event) {
+            // update continuous_dribbling_start_point once we start dribbling
+            continuous_dribbling_start_point = event.common.world.ball().position();
+            dribble(event);
+        };
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *get_possession_s + update_e[have_possession] / start_dribble = dribble_s,
```

Comment:
Noticed that other FSMs' naming conventions for actions have "update_" in front of them, for example, "update_start_dribble" or "update_get_possession", maybe should add the "update_" in these actions' names for consistency

Reply:
I don't think update_ adds much to helping the reader understand, so if I were to rewrite it, I'd remove the update_ from the action names